modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts

```diff
@@ -1052,38 +1058,54 @@ function DOMUtils(doc: Document, settings: Partial<DOMUtilsSettings> = {}): DOMU
 
   const createRng = () => doc.createRange();
 
-  const split = (parentElm: Node, splitElm: Node, replacementElm?: Node) => {
-    let r = createRng(), bef, aft, pa;
+  const split = <T extends Node>(parentElm: Node, splitElement: T, replacementElement?: T, optionsInput?: SplitOptions): T => {
+    const options: SplitOptions = optionsInput ? optionsInput : {};
+    let range = createRng();
+    let beforeFragment: DocumentFragment;
+    let afterFragment: DocumentFragment;
+    let parentNode: Node;
+
+    const doTrim = (fragment: Node, isBefore: boolean): Node => {
```

Comment:
There's a lot of duplication here... this could be simplified down. You could either split the "predicate" out into a new function, or just do a larger if statement. eg:

```
const trim = shouldTrim(fragment, options);
TrimNode.trimNode(self, fragment, !trim);
```
or
```
const trim = isEmpty(fragment) || options.skipTrimAfter && !isBefore || options.skipTrimBefore && isBefore;
TrimNode.trimNode(self, fragment, !trim);
```

Personally I prefer the first option, as it's cleaner to read.

Reply:
Makes sense. I admit the if-statement grew organically and I should have taken a step back and checked if it really had to look like that.